test/app-init/MfeDownloadManager.test.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import MfeDownloadManager from 'app-init/MfeDownloadManager';
+import { renderWithState } from 'test/testUtils';
+import { fetchMfeConfigList } from 'state/mfe/actions';
+
+jest.unmock('react-redux');
+
+jest.mock('state/mfe/actions', () => ({
+  fetchMfeConfigList: jest.fn(),
+}));
+
+describe('MfeDownloadManager', () => {
+  it('shows spinner when loading', () => {
+    renderWithState(<MfeDownloadManager>test</MfeDownloadManager>, { state: { currentUser: {} } });
+
+    expect(screen.getByTestId('spinner')).toBeInTheDocument();
+  });
+
+  it('shows startup wait screen when polling', async () => {
+    fetchMfeConfigList.mockImplementationOnce(() => () => new Promise(resolve => resolve({})));
+
+    renderWithState(<MfeDownloadManager>test</MfeDownloadManager>);
+
+    expect(await screen.findByTestId('startup-wait-screen')).toBeInTheDocument();
+  });
```

Comment:
I would add another test to check the same behavior when the result is `null` or `undefined` (it should happen when the API has a status code of 500).

Reply:
It seems that response won't ever be `null` or `undefined` since it comes from `fetch`. Also, when a `5xx` error occurs, the Promise gets rejected. So I just added a test for that scenario.